doc/source/whatsnew/v2.0.0.rst

```diff
@@ -341,6 +341,7 @@ Other API changes
 - Passing a sequence containing a type that cannot be converted to :class:`Timedelta` to :func:`to_timedelta` or to the :class:`Series` or :class:`DataFrame` constructor with ``dtype="timedelta64[ns]"`` or to :class:`TimedeltaIndex` now raises ``TypeError`` instead of ``ValueError`` (:issue:`49525`)
 - Changed behavior of :class:`Index` constructor with sequence containing at least one ``NaT`` and everything else either ``None`` or ``NaN`` to infer ``datetime64[ns]`` dtype instead of ``object``, matching :class:`Series` behavior (:issue:`49340`)
 - Changed behavior of :class:`Index` constructor with an object-dtype ``numpy.ndarray`` containing all-``bool`` values or all-complex values, this will now retain object dtype, consistent with the :class:`Series` behavior (:issue:`49594`)
+- Changed behavior of :class:`Index`, :class:`Series`, and :class:`DataFrame` arithmetic methods when working with object-dtypes, the results no longer do type inference on the result of the array operations (:issue:`??`)
```

Comment:
```suggestion
- Changed behavior of :class:`Index`, :class:`Series`, and :class:`DataFrame` arithmetic methods when working with object-dtypes, the results no longer do type inference on the result of the array operations and the resulting type will be ``object`` (:issue:`49714`)
```

?

Reply:
AFAICT that isnt 100% accurate, though we could take this a step further to make it work that way

Reply:
IMO `the resulting type will be object` would be a nicer, more predictable guarantee for the user. Would it be difficult to taking it a step further?

Reply:
im a bit skeptical of this for two reasons:

1) with 3rd party EAs we can't control what dtype they return (or we could, but it would be hacky)
2) for our own ops, there are a lot of places we do `result = np.array([op(x, y) for x, y in zip(self, other)])` where numpy will do inference.